docs/source/development/widget.md

```diff
@@ -251,9 +250,111 @@ See [Storybook](https://6.docs.plone.org/storybook) with available widgets.
 
 ## Write a new widget
 
-```{note}
-Please contribute to this section!
-```
+A custom widget is often used in Volto forms.
+To configure a new Volto widget follow these steps.
+
+1.  Add the following import as the first line in the file {file}`index.js` in the add-on to import the component of the actual widget.
+
+    ```jsx
+    import { ToggleWidget } from './ToggleWidget';
+    ```
+
+1.  Add the following line before returning from the `applyConfig` function inside the file {file}`index.js` in the add-on to register the widget add-on component.
+
+    ```jsx
+    config.widgets.widget.toggle_widget = ToggleWidget;
+    ```
+
+1.  Next, create the widget component file {file}`ToggleWidget.jsx` with the following content.
+
+    ```jsx
+    import { Button } from "semantic-ui-react";
+    export function ToggleWidget({ id, value, onChange }) {
+      return (
+        <Button
+          onClick={(e) => {
+            e.preventDefault();
+            onChange(id, value === "On" ? "Off" : "On");
+          }}
+        >
+          Toggle me: {value}
+        </Button>
+      );
+    }
+
+    export default ToggleWidget;
+    ```
+
+1.  Install `semantic-ui-react` as a dependency of your add-on to make it work.
```

Comment:
`semantic-ui-react` is already included in a Volto project, if you do not explicitly exclude it because you want to avoid it.

Reply:
Good point. We should not encourage the use of Semantic UI, now that it is deprecated. See https://6.docs.plone.org/volto/release-notes/index.html#id29 and an example at https://6.docs.plone.org/volto/theming/index.html.